src/dnf-plugins/product-id/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+CMAKE_MINIMUM_REQUIRED (VERSION 3.11.2)
+
+project(product-id C)
+
+# Warn user because CLion is upset by in source builds
+MACRO(MACRO_ENSURE_OUT_OF_SOURCE_BUILD MSG)
+    STRING(COMPARE EQUAL "${CMAKE_SOURCE_DIR}"
+    "${CMAKE_BINARY_DIR}" insource)
+    GET_FILENAME_COMPONENT(PARENTDIR ${CMAKE_SOURCE_DIR} PATH)
+    STRING(COMPARE EQUAL "${CMAKE_SOURCE_DIR}"
+    "${PARENTDIR}" insourcesubdir)
+    IF(insource OR insourcesubdir)
+        MESSAGE(FATAL_ERROR "${MSG}")
+    ENDIF(insource OR insourcesubdir)
+ENDMACRO(MACRO_ENSURE_OUT_OF_SOURCE_BUILD)
+
+MACRO_ENSURE_OUT_OF_SOURCE_BUILD(
+"${CMAKE_PROJECT_NAME} requires an out of source build. Delete all CMake generated files or CLion won't work."
+)
+
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_STANDARD 11)
+
+#ADD_COMPILE_OPTIONS(-Wall -fPIC -g -Wl,as-needed)
+
+# Build type
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Debug' as none was specified.")
+    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
+endif ()
+
+if (CMAKE_COMPILER_IS_GNUCC)
+    set (CMAKE_C_FLAGS "-Wall -fPIC -Wextra -pedantic -Wno-long-long")
```

Comment:
With `-pedantic`, I see some warnings like

```
 warning: ISO C99 requires at least one argument for the "..." in a variadic macro
                 error("Product certificate does not contain required ID");
```

according to [SE](https://stackoverflow.com/questions/43991106/iso-c99-requires-at-least-one-argument-for-the-in-a-variadic-macro-when#43991526) that warning still applies in C11.

Reply:
Strange. I don't see this warning.